public/app/plugins/datasource/graphite/datasource.ts

```diff
@@ -750,35 +751,20 @@ export class GraphiteDatasource
     const httpOptions = {
       method: 'GET',
       url: '/functions',
+      // add responseType because if this is not defined, backend_srv defaults to json
+      responseType: 'text',
     };
 
     return lastValueFrom(
       this.doGraphiteRequest(httpOptions).pipe(
         map((results: any) => {
-          if (results.status !== 200 || typeof results.data !== 'object') {
-            if (typeof results.data === 'string') {
-              // Fix for a Graphite bug: https://github.com/graphite-project/graphite-web/issues/2609
-              // There is a fix for it https://github.com/graphite-project/graphite-web/pull/2612 but
-              // it was merged to master in July 2020 but it has never been released (the last Graphite
-              // release was 1.1.7 - March 2020). The bug was introduced in Graphite 1.1.7, in versions
-              // 1.1.0 - 1.1.6 /functions endpoint returns a valid JSON
-              const fixedData = JSON.parse(results.data.replace(/"default": ?Infinity/g, '"default": 1e9999'));
-              this.funcDefs = gfunc.parseFuncDefs(fixedData);
-            } else {
-              this.funcDefs = gfunc.getFuncDefs(this.graphiteVersion);
-            }
-          } else {
-            this.funcDefs = gfunc.parseFuncDefs(results.data);
-          }
-
-          // When /functions endpoint returns application/json response but containing invalid JSON the fix above
-          // wont' be triggered due to the changes in https://github.com/grafana/grafana/pull/45598 (parsing happens
-          // in fetch and Graphite receives an empty object and no error). In such cases, when the provided JSON
-          // seems empty we fallback to the hardcoded list of functions.
-          // See also: https://github.com/grafana/grafana/issues/45948
-          if (Object.keys(this.funcDefs).length === 0) {
```

Comment:
@Dieterbe is there a case when graphite API itself would return empty list of functions and it would make sense for Grafana to supply some defaults?

This here is removed because because we circumvent the changed behaviour that invalidated the workaround and we always get the response as text and we can then apply workaround for the `infinity` issue. But that means if Graphite returns empty list of functions we assume there are no functions in Graphite. Is that valid or will Graphite always return at least the default function set?

Reply:
Graphite should never return an empty list of functions, unless something went badly wrong with it :) 
Note: ownership of our graphite product has been moved to the query squad. i'm no longer involved.